app/models/user.rb

```diff
@@ -327,6 +329,31 @@ def active_for_authentication?
 
   def confirm!
     super
+
+    # send MailChimp subscription data
+    if self.email_subscriber
```

Comment:
Because this is repeat of what is in the controller. This code should be moved into common place.  I'd suggest putting it in `/lib`. A recent example is `lib/signed_jwt.rb`.